### ISSUE TITLE
Add def for only playoffs games schedule

### ISSUE THREAD

**Issue by Natan7** [jaebradley/basketball_reference_web_scraper#263]
Maybe it has an interesting function for getting ONLY playoffs matches and results, because some things change in playoffs, like competitiveness, home court crowd pressure, among other attributes. I can try to code this feature, if you validate it, like cool add.

**Comment by Natan7** [comment]
Another interesting function would be a getter of advanced statistics (teams rating page on Basketball Reference).

**Comment by jaebradley** [comment]
>Another interesting function would be a getter of advanced statistics (teams rating page on Basketball Reference).

@Natan7 are you talking about this table

<img width="1444" alt="image" src="https://github.com/jaebradley/basketball_reference_web_scraper/assets/8136030/448a99d4-9aff-4611-9164-092080eff32c">

From https://www.basketball-reference.com/leagues/NBA_2024.html

**Comment by Natan7** [comment]
> > Another interesting function would be a getter of advanced statistics (teams rating page on Basketball Reference).
> 
> @Natan7 are you talking about this table
> <img alt="image" width="1444" src="https://private-user-images.githubusercontent.com/8136030/319005875-448a99d4-9aff-4611-9164-092080eff32c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTIxNjQzMzMsIm5iZiI6MTcxMjE2NDAzMywicGF0aCI6Ii84MTM2MDMwLzMxOTAwNTg3NS00NDhhOTlkNC05YWZmLTQ2MTEtOTE2NC0wOTIwODBlZmYzMmMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDQwMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA0MDNUMTcwNzEzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZDM5YTMyYjY4MzNhNDMwMGQ0MmJmNjBkNDk1OWYyNzEwZjMyYWY5N2U1Yzc5Nzg1ODMwM2ZhOWZmYTE5ZjE3NyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.2rSrOELCIuOMD1r8J9FoBn5VqXbuJ31InNNk9jkk1vc">
> 
> From https://www.basketball-reference.com/leagues/NBA_2024.html

Yes and this table -> https://www.basketball-reference.com/leagues/NBA_2023_ratings.html

**Comment by Natan7** [comment]
> Maybe it has an interesting function for getting ONLY playoffs matches and results, because some things change in playoffs, like competitiveness, home court crowd pressure, among other attributes. I can try to code this feature, if you validate it, like cool add.

@jaebradley this table for playoff issue -> https://www.basketball-reference.com/playoffs/NBA_2023_games.html

**Comment by DigitalCurrensy** [comment]
Happy to pick this up if it's still wanted.

Proposed approach: a new `client.playoff_season_schedule(season_end_year)` backed by the dedicated playoff page @Natan7 linked above (`/playoffs/NBA_{year}_games.html`). That page uses the same `table#schedule` markup and the same `data-stat` attributes as the monthly schedule pages, so `SchedulePage`, `ScheduleRow`, `parse_scheduled_games`, and `SCHEDULE_COLUMN_NAMES` all get reused unchanged. Purely additive — no change to `season_schedule` or its output.

One reason I'd lean toward that page rather than deriving playoff games from the monthly pages: `SchedulePage.rows` already drops the `"Playoffs"` divider row, and that divider only appears in the month the playoffs begin — the May and June pages are entirely playoff games with no marker at all. So deriving it would need stateful traversal across months, and it'd be ~10 requests instead of 1. Given the rate-limit concerns you raised in #284, the single-page route seemed better.

Two questions before I write anything:

1. Separate function, or would you rather it were a `playoffs=True` style parameter on `season_schedule`?
2. For the integration test, is a saved copy of the real page as a fixture fine (matching the existing `tests/integration/files/schedule/` convention, ~220KB), or would you prefer a smaller hand-built fixture?

Also worth flagging: that page starts at Round 1 Game 1, so play-in games aren't included. I'd document that explicitly either way.

**Comment by jaebradley** [comment]
@DigitalCurrensy hopefully, the below makes sense

>the single-page route seemed better.

I concur 👍 

>Separate function, or would you rather it were a playoffs=True style parameter on season_schedule?

I have a preference for a separate function. `season_schedule` already returns these games, and a flag would need to be added to any output to indicate they were playoff games. While slightly more annoying to the client user if the client user only wanted to identify regular season games (as they'd need to use both methods)

>For the integration test, is a saved copy of the real page as a fixture fine (matching the existing tests/integration/files/schedule/ convention, ~220KB), or would you prefer a smaller hand-built fixture?

This is a good question. In reality, a smaller fixture file that strips out irrelevant JavaScript is probably best. However, obviously, if you've introspected the integration test files, you'll notice this is not what is done in practice. I need to formalize and evolve some of the integration test practices in the not-so-distant future. **tl;dr:** If you include a smaller HTML-only fixture file for the page, that would be preferred.
